### DISCUSSION TITLE
[Unknown Mod][Game] Hero portraits (small and big), display of the hero's state when he is "sleeping" and "at sea"

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4763] [Ideas]
**All images are for explanation only, the position of the icons and the icons used can be changed.**

In fheroes2, it is possible to change the state of the hero to "sleeping" with the "z" key on the keyboard.
In this state the hero is ignored when the user uses the buttons "Next Hero" and "End turn".

This state is indicated by a light blue diamond positioned over the top of the green bar of the hero's remaining spell points (for the small hero portrait), which is not ideal.

Jezebel with the light blue diamond icon to indicate the sleeping hero.
![Hero sleeping state 01](https://user-images.githubusercontent.com/43583994/145212927-559d3475-91ec-4a85-a274-ec18555be841.png)

I suggest to position the icon of this state a little bit to the left on the top right hero portrait graphic in order to free up the view of the green bar of the hero's remaining spell points.
At the same time, the icon should be changed to be more explicit and also smaller so as not to hide the hero's portrait too much.

Jezebel with the "Z" icon to indicate the sleeping hero (and at the same time to indicate the "z" key on the keyboard that allows to activate and deactivate the hero's sleeping state).
![Hero small portrait Z 01](https://user-images.githubusercontent.com/43583994/145213034-87668d3e-fc93-442f-8ace-f01aad1018a3.png)

In fheroes2, there is the option "castle: allow gardians" which allows to put a hero in garrison by clicking on the Captain's slot when a hero is in visit.
This hero takes the place of the Captain in the castle.

Arie is now a guardian hero.
![Hero guardian 01](https://user-images.githubusercontent.com/43583994/145215004-fb341142-189b-40e9-a022-1b137399bd11.png)

Same problem as for the previous case, this state is indicated by an orange shield placed over the green bar of the hero's remaining spell points (for the small hero portrait).

Tyro with the orange shield icon.
![Zzz Castle Boat icons 002](https://user-images.githubusercontent.com/43583994/145215169-75a24420-6400-44ef-bee9-4ed8573385c4.png)

Troyan with both the sleeping and guardian icons.
![Hero guardian sleeping 01](https://user-images.githubusercontent.com/43583994/145215245-7d65337d-9f45-4d0e-9c4d-9d2a1c9fc111.png)

Same problem, same solution, I also suggest to position the icon of this state a little bit to the left on bottom right the hero portrait graphic in order to free up the view of the green bar of the hero's remaining spell points.
At the same time, the icon could be changed to be more representative (why the choice of the orange color for all players?) and also smaller so as not to hide the hero's portrait too much.

Jezebel with both the new sleeping and guardian icons.
![Hero small portrait Z castle 01](https://user-images.githubusercontent.com/43583994/145217304-e9ec2a96-766e-4bf1-bad5-e9ac923df749.png)

In the previous versions of fheroes2, there was also an icon displayed when a hero was on a boat at sea.
This feature has been removed, I don't really know why, because it was a good way to spot at a quick glance which heroes were at sea.

I suggest to reintroduce this handy feature. :)

Jezebel sleeping in a boat at sea (small portrait). :D
![Hero small portrait Z boat 03](https://user-images.githubusercontent.com/43583994/145217720-521f26ad-5314-473a-a612-53ea00dc6e68.png)

The icons are also too small in the screens with a big hero portrait.

Tyro with the orange shield icon (big portrait).
![Zzz Castle Boat icons 003](https://user-images.githubusercontent.com/43583994/145220212-1547e4cc-0c5e-4169-bf2c-cfad5de0522c.png)

Jezebel sleeping in a boat at sea (big portrait with bigger icons).
![Hero Zzz Boat 01](https://user-images.githubusercontent.com/43583994/145218227-5b49b600-4ed2-4e8f-8258-7fc9d86b06c6.png)

Jezebel sleeping as guardian (big portrait with bigger icons).
![Hero Zzz Castle 01](https://user-images.githubusercontent.com/43583994/145218236-e220482a-acf9-4b57-867c-e9a4900b3ad8.png)

This is not an option.
This can be implemented directly in fheroes2.

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu 
I think, "sleeping" heroes could have a grayscale portraits in a hero list, indicating their state. Small icons at high resolutions will be hardly distinguishable. ;)

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> Hi, @LeHerosInconnu I think, "sleeping" heroes could have a grayscale portraits in a hero list, indicating their state. Small icons at high resolutions will be hardly distinguishable. ;)

I already have other proposals planned. :)
Stay tuned. :D

**Reply by LeHerosInconnu** [reply]
> I already have other proposals planned. :)
> Stay tuned. :D

Why not something like this?
For the "sleeping" hero, a graphic representing night is displayed at the top of the portrait (in the example, it's more like early morning :) ).
For the hero in a boat, a graphic representing an element of a boat is displayed at the bottom of the portrait (in the example, it's a guardrail).
Images are for explanation only.

![Hero on boat sleeping 03](https://github.com/ihhub/fheroes2/assets/43583994/9be93a24-3bd9-4c37-9ac7-de62176bc098)

![Hero on boat sleeping big 02](https://github.com/ihhub/fheroes2/assets/43583994/b8376a77-3ef9-4dc8-b9d9-7dd6e64682b2)

**Reply by zenseii** [reply]
How romantic Lord Kilburn is looking at us with the sunrise in the background 😍

**Comment by Mr-Bajs** [comment]
The greyscale approach is probably the best solution.

**Comment by LeHerosInconnu** [comment]
Alternatively, it is possible to change the color of the movement bar to blue and/or add the symbol of a boat anchor into the movement bar when the hero is in a boat.
For sleeping heroes, the spell points bar can be changed to purple and/or the symbol of a quarter moon can be added to the spell points bar.

![Hero on boat sleeping 03](https://github.com/ihhub/fheroes2/assets/43583994/4273e9c3-aa06-4bb8-890a-6ec3aa08e5f4)

For the larger hero portrait, bigger boat anchor and quarter-moon symbols are used.

**Edit 2023/10/01.**

Image is for explanation only. :)

**Reply by Branikolog** [reply]
I personally love the idea of changing the bar colour for sailing hero. Indeed, hero has two independent movement scales for land and ocean. I think it would be visually more pleasant, if while sailing hero has a slightly different colour for movement bar. I'm not sure only for the colour proposed by @LeHerosInconnu , but we have 253 more to choose. ;)

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> ...I'm not sure only for the colour proposed by @LeHerosInconnu , but we have 253 more to choose. ;)

Oh, I forgot to mention that the image is for explanation only.
I've edited the post. :D

**Comment by roman-yagodin** [comment]
Personally, I think some kind of 💤 icon would be best choice to show hero sleeping state. 

As for controls, there are pretty much free space in hero dialog for some additional buttons:

![sleeping_hero-1](https://github.com/user-attachments/assets/e4cee7c8-e020-43ff-86ac-7f18417d94e5)

**Comment by LeHerosInconnu** [comment]
This has been remanied, here: https://github.com/ihhub/fheroes2/discussions/10089.
I close the discussion.
